### ISSUE TITLE
git resource cannot resolve host when using Docker for Mac

### ISSUE THREAD

**Issue by chenbh** [concourse/concourse#6038]
Tldr: `concourse/git-resource` is broken for Docker on Mac and we should use the ubuntu versions (e.g. docker compose with `concourse/concourse:ubuntu` or mirror `concourse/git-resource:ubuntu` as a custom resource type)

I'm using curl for all of the examples here, but the same thing occurs with git.

## Problem
So we use `alpine:edge` as the base for all of our resources and it seems like the latest version of `musl` ([1.2.1-r0](https://pkgs.alpinelinux.org/packages?name=musl&branch=edge)) somehow made it so that it no longer tries IPv4 lookup if IPv6 fails (maybe as part of [this fix?](https://git.musl-libc.org/cgit/musl/tree/WHATSNEW#n2223)). 
Unfortunately IPv6 is [not supported](https://docs.docker.com/docker-for-mac/troubleshoot/#known-issues) on Docker for Mac and so all DNS queries will fail to resolve if musl 1.2.1-r0 is installed.

```
$ docker run -it alpine:edge
# apk add curl
...
# curl https://github.com
<html>...</html>                                        <- works
# apk add --upgrade musl
(1/1) Upgrading musl (1.1.24-r9 -> 1.2.1-r0)
# curl https://github.com
curl: (6) Could not resolve host: github.com            <- breaks
```
Doing the exact same steps above but with `alpine:3` (which only supports musl 1.1.24-r9) works perfectly fine.

## Steps to reproduce

While the git resource would normally be created by Guardian as a container-in-container, the same effect still occurs if we run it as a normal Docker container:
```
$ docker pull concourse/git-resource
$ docker run -it concourse/git-resource
/usr/lib # curl https://github.com
curl: (6) Could not resolve host: github.com
```

## Workaround
The ubuntu version of git-resource is still working fine and comes packaged with `concourse/concourse:ubuntu` (or `6-ubuntu`, `6.5-ubuntu`, etc). Changing it at the `docker-compose.yml` level should fix it

Alternatively we can define the ubuntu resource as a resource type and use it at the pipeline level.
```
resource_types:
- name: git-ubuntu
  type: registry-image
  source: 
    repository: concourse/git-resource
    tag: ubuntu

resources:
- name: concourse
  type: git-ubuntu
  source:
    uri: https://github.com/concourse/concourse.git
```

## Fix
As far as I can tell only the git-resource is broken: `alpine:edge` still comes with musl 1.1.24 and only git-resource upgrades it to 1.2.1 (inadvertently as part of [installing g++](https://github.com/concourse/git-resource/blob/3c68b5f8a5a6974d277cb856b8e3f16443f4ec80/dockerfiles/alpine/Dockerfile#L15))

@vito couple questions:
- Should we move off edge and start using `alpine:3`? It seems to be going steady since 2014.
- Should we use a custom base image for our resources? That way we wouldn't need to go around and change a bunch of Dockerfiles when something like this comes up

## Tirage

Concourse version 6.5.1 (probably also affects 6.5.0)
Git resource version: 1.9.0?

**Comment by chenbh** [comment]
Update: it appears `alpine:edge` now ships with musl 1.2.1 and is just straight up broken on Mac
```
$ docker run -it alpine:edge
# apk list musl
musl-1.2.1-r1 x86_64 {musl} (MIT) [installed]
# apk add curl
# curl https://github.com
curl: (6) Could not resolve host: github.com
```

**Comment by vito** [comment]
related:

* https://gitlab.alpinelinux.org/alpine/aports/-/issues/11879
* https://github.com/docker/for-mac/issues/5020

sounds like there's a fix in Docker for Mac coming in v3.1.

**Comment by stephen-turner** [comment]
Docker Desktop 3.1.0 is now released. Please test whether it fixes your bug.

**Comment by vito** [comment]
@stephen-turner Great! We'll try it out shortly. Thanks for the poke!

**Comment by maggie0002** [comment]
> Docker Desktop 3.1.0 is now released. Please test whether it fixes your bug.

FYI I was experiencing this issue resolving codeload.github.com when using MacBook. Updated Docker, now running 3.1.0 and the problem appears to be resolved.

**Comment by vito** [comment]
Haven't had time to confirm this myself, but going to tentatively close.

**Comment by jutkko** [comment]
Have hit the issue and can confirm that Docker Desktop 3.1.0 fixes my issue. Thanks!

**Comment by danmcadams** [comment]
Fixed it for me, thanks!

**Comment by averypmc** [comment]
Ditto 🎉

**Comment by RammusXu** [comment]
This happens on `docker desktop: 2.5.0.1`.
It's fixed after I upgrade to `3.2.2`

**Comment by matteocollina** [comment]
Yes fixed after upgrade to `3.2.2`!

**Comment by jutkko** [comment]
Is this broken again? I am getting this error with most up to date Docker for Mac 3.5.2.

**Comment by douglasjose** [comment]
I'm having a similar issue after upgrading to Concourse 7.4. I can consistently reproduce the issue with this version; downgrading to `concourse/concourse:7.2` or `concourse/concourse:7.3` fixes the problem.

```
Identity added: /tmp/git-resource-private-key (Concourse CI SSH key)
Cloning into '/tmp/git-resource-repo-cache'...
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

I'm running Docker Desktop 3.6.0 on MacOS. Should this issue be reopened?
